pygsti/protocols/vb.py

```diff
@@ -67,6 +68,25 @@ def map_qubit_labels(self, mapper):
         mapped_qubit_labels = self._mapped_qubit_labels(mapper)
         return ByDepthDesign(self.depths, mapped_circuit_lists, mapped_qubit_labels, remove_duplicates=False)
 
+    def truncate_to_lists(self, list_indices_to_keep):
+        """
+        Truncates this experiment design by only keeping a subset of its circuit lists.
+
+        Parameters
+        ----------
+        list_indices_to_keep : iterable
+            A list of the (integer) list indices to keep.
+
+        Returns
+        -------
+        ByDepthDesign
+            The truncated experiment design.
+        """
+        ret = _copy.deepcopy(self) # Works for derived classes too
+        ret.depths = [self.depths[i] for i in list_indices_to_keep]
+        ret.circuit_lists = [self.circuit_lists[i] for i in list_indices_to_keep]
+        return ret
```

Comment:
Might there be scenarios where the deepcopy isn't needed? If so, it might be good to have an optional function argument ``deepcopy=False`` that controls whether the copy is deep or shallow.

Reply:
There could be cases where the deepcopy is not desired - for example, if we want to create a sort of "nested" edesign and want to point to the original circuit lists.
However, I'm inclined to keep this deepcopy as-is for now because it's actually indicative of a hack to make these functions work for derived classes. I think we could more cleanly do this with Generics/TypeVars, which I've been learning more about recently, but I want to keep this the same as the pattern used elsewhere in the code so I can clean it up all at once later and don't feel like plumbing a temporary kwarg in through all of this now.